SamplesDashboard/SamplesDashboard/Services/SampleService.cs

```diff
@@ -106,17 +111,25 @@ public async Task<List<Node>> GetSamples()
         /// <returns> A list of samples.</returns>
         private async Task SetHeadersAndStatus(Node sampleItem) 
         {
-            var headerDetails = await GetHeaderDetails(sampleItem.Name);
-            sampleItem.Language = headerDetails.GetValueOrDefault("languages");
-            sampleItem.FeatureArea = headerDetails.GetValueOrDefault("services");
-
-            if (!_cache.TryGetValue(sampleItem.Name, out Repository repository)) 
+            
+            Repository repository;
+            if (!_cache.TryGetValue(sampleItem.Name, out repository))
             {
                 repository = await GetRepository(sampleItem.Name);
+
+                if (repository.DependencyGraphManifests?.Nodes?.Length > 0)
+                {
+                    sampleItem.HasDependendencies = true;
+                }
+
                 var cacheEntryOptions = new MemoryCacheEntryOptions().SetAbsoluteExpiration(TimeSpan.FromSeconds(_config.GetValue<double>("timeout")));
                 _cache.Set(sampleItem.Name, repository, cacheEntryOptions);
-            }
+            }    
+            
             sampleItem.SampleStatus = repository.highestStatus;
+            var headerDetails = await GetHeaderDetails(sampleItem.Name);
```

Comment:
Since we already know that we won't display those without dependencies, we could skip fetching header details when `HasDependendencies` is true.